registry/handlers/app.go

```diff
@@ -194,6 +195,28 @@ func NewApp(ctx context.Context, config *configuration.Configuration) *App {
 		}
 	}
 
+	// configure foreign layers
+	if flConfig, ok := config.Storage["foreignlayers"]; ok {
+		if hosts, ok := flConfig["hostwhitelist"]; ok {
+			switch hosts := hosts.(type) {
+			case []interface{}:
```

Comment:
IMHO supporting that variation is out of scope. We should test to ensure something like `REGISTRY_STORAGE_FOREIGNLAYER="hostwhitelist: [a.com]"` is possible, though.